lean/imap.lean

```diff
@@ -0,0 +1,33 @@
+/- A simple interval map -/
+
+structure {u v} data.imap.imap_entry (k : Type u) (val : Type v) : Type (max u v) := 
+  (start : k) 
+  (extent : k)
+  (value : val)
+
+def {u v} data.imap (k : Type u) (val : Type v) := list (data.imap.imap_entry k val)
```

Comment:
Shouldn't ordering be part of type for this and `imap_entry` as in `rb_map`?

Reply:
Test comment

Reply:
I added the lt param to the type of imap, but we might want to figure out whether we want some more complicated datastructure before following rbtree/rbmap